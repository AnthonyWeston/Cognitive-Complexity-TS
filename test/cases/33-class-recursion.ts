class NewSelf {
    constructor() {
        new NewSelf();
    }
}

class RecursiveConstructor {
    constructor() {
        this.constructor();
    }
}

const anonymousClass = class {
    method() {
        new anonymousClass();
    }
}

const nameForClass = class NamedClass {
    alias() {
        new nameForClass();
    }

    realName() {
        new NamedClass();
    }
}
