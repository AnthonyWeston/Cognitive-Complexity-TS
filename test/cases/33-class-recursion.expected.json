{
    "33-class-recursion.ts": {
        "score": 5,
        "inner": [
            {
                "name": "NewSelf",
                "score": 1,
                "line": 1,
                "column": 1,
                "inner": [
                    {
                        "name": "constructor",
                        "score": 1,
                        "line": 2,
                        "column": 5
                    }
                ]
            },
            {
                "name": "RecursiveConstructor",
                "score": 1,
                "line": 7,
                "column": 1,
                "inner": [
                    {
                        "name": "constructor",
                        "score": 1,
                        "line": 8,
                        "column": 5
                    }
                ]
            },
            {
                "name": "anonymousClass",
                "score": 1,
                "line": 13,
                "column": 24,
                "inner": [
                    {
                        "name": "method",
                        "score": 1,
                        "line": 14,
                        "column": 5
                    }
                ]
            },
            {
                "name": "NamedClass",
                "score": 2,
                "line": 19,
                "column": 22,
                "inner": [
                    {
                        "name": "alias",
                        "score": 1,
                        "line": 20,
                        "column": 5
                    },
                    {
                        "name": "realName",
                        "score": 1,
                        "line": 24,
                        "column": 5
                    }
                ]
            }
        ]
    }
}
